From e1a14c1f6d0bef227a4b1b848b4172f6a172c0f7 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 31 Mar 2020 19:04:37 +0800
Subject: [PATCH] net: wireless: rockchip: rkwifi: bcmdhd: fix gcc warning

Fix gcc 9 warning:

drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c: In function 'dhd_attach':
drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c:9494:4: warning: 'strncat' specified bound 2 equals source length [-Wstringop-overflow=]
 9494 |    strncat(if_name, "%d", 2);
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~

drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmutils.c: In function 'bcm_format_flags':
drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmutils.c:2140:3: warning: 'strncpy' specified bound depends on the length of the source argument [-Wstringop-overflow=]
 2140 |   strncpy(p, name, nlen + 1);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~

drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmutils.c: In function 'bcm_format_field':
drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmutils.c:2097:5: warning: 'strncpy' specified bound depends on the length of the source argument [-Wstringop-overflow=]
 2097 |     strncpy(buf, name, slen+1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~

Change-Id: Iab941c1789cfa0e3a1322cdbf64fcfbb75bd9c7c
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmutils.c  | 4 ++--
 drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmutils.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmutils.c
index 1ba8481c50ec..1a0715e49905 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmutils.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmutils.c
@@ -2094,7 +2094,7 @@ bcm_format_field(const bcm_bit_desc_ex_t *bd, uint32 flags, char* buf, int len)
 		if ((flags & mask) == bit) {
 			if (len > (int)strlen(name)) {
 				slen = strlen(name);
-				strncpy(buf, name, slen+1);
+				strncpy(buf, name, len);
 			}
 			break;
 		}
@@ -2137,7 +2137,7 @@ bcm_format_flags(const bcm_bit_desc_t *bd, uint32 flags, char* buf, int len)
 		if (len <= slen)
 			break;
 		/* copy NULL char but don't count it */
-		strncpy(p, name, nlen + 1);
+		strlcpy(p, name, nlen + 1);
 		p += nlen;
 		/* copy btwn flag space and NULL char */
 		if (flags != 0)
diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
index c366cb4c6256..81da661da2a5 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_linux.c
@@ -9491,7 +9491,7 @@ dhd_attach(osl_t *osh, struct dhd_bus *bus, uint bus_hdrlen
 		len = strlen(if_name);
 		ch = if_name[len - 1];
 		if ((ch > '9' || ch < '0') && (len < IFNAMSIZ - 2))
-			strncat(if_name, "%d", 2);
+			strlcat(if_name, "%d", IFNAMSIZ);
 	}
 
 	/* Passing NULL to dngl_name to ensure host gets if_name in dngl_name member */
-- 
2.35.3

